xwiki-platform-core/xwiki-platform-webjars/xwiki-platform-webjars-api/src/main/java/org/xwiki/webjars/internal/WebJarsResourceReferenceHandler.java

```diff
@@ -90,27 +97,79 @@ protected String getResourceName(WebJarsResourceReference resourceReference)
     @Override
     protected boolean isResourceCacheable(WebJarsResourceReference resourceReference)
     {
-        return !Boolean.valueOf(resourceReference.getParameterValue("evaluate"));
+        return !Boolean.parseBoolean(resourceReference.getParameterValue("evaluate"));
     }
 
     @Override
-    protected InputStream filterResource(WebJarsResourceReference resourceReference, InputStream resourceStream)
+    protected InputStream filterResource(WebJarsResourceReference resourceReference, InputStream resourceStream,
+        HttpServletResponse response)
         throws ResourceReferenceHandlerException
     {
+        String resourceName = getResourceName(resourceReference);
+        InputStream stream;
         if (!isResourceCacheable(resourceReference)) {
-            String resourceName = getResourceName(resourceReference);
-            try {
-                // Evaluates the given resource using Velocity.
-                StringWriter writer = new StringWriter();
-                this.velocityManager.getVelocityEngine().evaluate(this.velocityManager.getVelocityContext(), writer,
-                    resourceName, new InputStreamReader(resourceStream, UTF8));
-                return new ByteArrayInputStream(writer.toString().getBytes(UTF8));
-            } catch (Exception e) {
-                throw new ResourceReferenceHandlerException(
-                    String.format("Failed to evaluate the Velocity code from WebJar resource [%s]", resourceName), e);
+            if (resourceName.endsWith(".less")) {
+                stream = lessResourceHandler(resourceReference, resourceStream, response);
+            } else {
+                stream = defaultResourceHandler(resourceStream, resourceName);
```

Comment:
WDYT of moving these two calls to dedicated internal components?

Reply:
+1 to introduce a ``WebJarsResourceFilter`` component with two implementations ("velocity" and "less").

Reply:
see 
- the role: https://github.com/xwiki/xwiki-platform/pull/1586/files#diff-16deb1721a8b7b94d4d569c7592eedd1b3791da4bba55054b5cfc6e71e02e50bR35
- the less component: https://github.com/xwiki/xwiki-platform/pull/1586/files#diff-5a5014f40184c0b97b9f3a265b874c94aa5a8eb26cb713872f0cfea5f05600bdR48
- the velocity component: https://github.com/xwiki/xwiki-platform/pull/1586/files#diff-259a2897677e7d165d728f64a24176c9b19af4450037396d41349f93eeb01d16R46